[Dashboard as code] Embeddable API Integration

ThomThomson opened the pull request:
Closes https://github.com/elastic/kibana/issues/192622
Closes https://github.com/elastic/kibana/issues/222472
Closes https://github.com/elastic/kibana/issues/222466

nickpeihl committed:
[Dashboards as Code] Stub out embeddable content management registry (#221692)

Note: _This PR does not need to be reviewed by external teams. This PR
merges into a feature branch that Kibana presentation team is working on
to support embeddable integrations with Dashboards as Code._

Commit:
Add embeddabEmbeddableStart to dashboard transforms, move inject/extract reference handling to server

Commit:
Add embeddable cmd to panel transforms

Commit:
Update example embeddables with CMDs

Commit:
Make book SavedObject schema absolutely irredeemably stupid-looking

Commit:
Defer panel migration for eEmbeddableStart availability

nickpeihl committed:
[Dashboards as Code] Register example book saved object and content management (#221981)

Note: _This PR does not need to be reviewed by external teams. This PR
merges into a feature branch that Kibana presentation team is working on
to support embeddable integrations with Dashboards as Code._

Commit:
Remove unneeded flow call

nickpeihl committed:
[Dashboards as Code] Provide embeddable service for migration (#222059)

Adds a `startServices` method for the dashboard server to provide the
embeddable service that can be used in the migration to correctly
extract references.

This also logs a warning instead of throwing if the migration fails.

Here is an example dashboard saved object that triggers the migration on
import.

[7.8.0-dashboard.ndjson.zip](https://github.com/user-attachments/files/20521831/7.8.0-dashboard.ndjson.zip)

---------

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>

kibanamachine commented:
<!-- kbn-message-context:project-deploy-job -->
Project deployments require a Github label, please add one or more of `ci:project-deploy-(elasticsearch|observability|security)` and trigger the job through the checkbox again.

Commit:
Merge branch 'dashboardApi/embeddableIntegration' of https://github.com/elastic/kibana into dashboardApi/embeddableIntegration

Commit:
Save and load book embeddables from SavedBookStorage

Commit:
Make saved books visible in management

Commit:
Add book embeddable to Add from Library

Commit:
Add sequelTo saved object referenceobject to book example

Commit:
Remove attriubutes and references from item schemas

Commit:
Fix panel in transform

Commit:
Deprectate references in SerializedPanelState

Commit:
Fix deprecation comment

nickpeihl committed:
Merge remote-tracking branch 'upstream/main' into dashboardApi/embeddableIntegration

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/notice'

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/305973)
* Commit: 43b73490a7579f4f3139b2b4aecdc3b684fab46a

### Failed CI Steps
* [Check Types](https://buildkite.com/elastic/kibana-pull-request/builds/305973#019741d5-7cc3-4b4c-9d71-0f021e2b103a)

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/305973","number":305973,"commit":"43b73490a7579f4f3139b2b4aecdc3b684fab46a"}],"number":305973}
buildkite-pr-comment-->

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/eslint --no-cache --fix'

nickpeihl committed:
[Dashboards as Code][Embeddable API Integration] Make EmbeddableContentManagementRegistry async (#223050)

Note: _This PR does not need to be reviewed by external teams. This PR
merges into a feature branch that Kibana presentation team is working on
to support embeddable integrations with Dashboards as Code._

Makes the `EmbeddableContentManagementRegistry` an async registry to
reduce bundle size.

Commit:
Make extractDashboardState async and pull from CM registry

Commit:
Fix running savedObjectToItem on pre-8.17 panels

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/eslint --no-cache --fix'

Commit:
Fix typecheck and i18n

Commit:
Merge branch 'dashboardApi/embeddableIntegration' of https://github.com/elastic/kibana into dashboardApi/embeddableIntegration

nickpeihl committed:
Update unit tests for async functions

Commit:
Fix unused ts-expect-error

Commit:
Merge branch 'dashboardApi/embeddableIntegration' of https://github.com/elastic/kibana into dashboardApi/embeddableIntegration

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/307091)
* Commit: dd2d7d33cf04fd0cb9aaac8d5915dbc4982a3c43

### Failed CI Steps
* [Check Types](https://buildkite.com/elastic/kibana-pull-request/builds/307091#01975a98-c8c7-4fec-be2d-37d581d7e72b)

### History
* :broken_heart: [Build #306779](https://buildkite.com/elastic/kibana-pull-request/builds/306779) failed f4146b455542142f35c50d27777d1c40ae3e4e92
* :broken_heart: [Build #306693](https://buildkite.com/elastic/kibana-pull-request/builds/306693) failed 06daa5e6cfd246409ec19f848594d798e3826a1a
* :broken_heart: [Build #305973](https://buildkite.com/elastic/kibana-pull-request/builds/305973) failed 43b73490a7579f4f3139b2b4aecdc3b684fab46a

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/307091","number":307091,"commit":"dd2d7d33cf04fd0cb9aaac8d5915dbc4982a3c43"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/306779","number":306779,"commit":"f4146b455542142f35c50d27777d1c40ae3e4e92"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/306693","number":306693,"commit":"06daa5e6cfd246409ec19f848594d798e3826a1a"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/305973","number":305973,"commit":"43b73490a7579f4f3139b2b4aecdc3b684fab46a"}],"number":307091}
buildkite-pr-comment-->

Commit:
Remove @ts-expect-errors

Commit:
Merge remote-tracking branch 'upstream/main' into dashboardApi/embeddableIntegration

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/307165)
* Commit: f00b67e9b3abcbbcba68afce01050788804d463f

### Failed CI Steps
* [Check Types](https://buildkite.com/elastic/kibana-pull-request/builds/307165#01975b5e-79af-4a26-8e2f-d546f7605051)

### History
* :broken_heart: [Build #307091](https://buildkite.com/elastic/kibana-pull-request/builds/307091) failed dd2d7d33cf04fd0cb9aaac8d5915dbc4982a3c43
* :broken_heart: [Build #306779](https://buildkite.com/elastic/kibana-pull-request/builds/306779) failed f4146b455542142f35c50d27777d1c40ae3e4e92
* :broken_heart: [Build #306693](https://buildkite.com/elastic/kibana-pull-request/builds/306693) failed 06daa5e6cfd246409ec19f848594d798e3826a1a
* :broken_heart: [Build #305973](https://buildkite.com/elastic/kibana-pull-request/builds/305973) failed 43b73490a7579f4f3139b2b4aecdc3b684fab46a

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/307165","number":307165,"commit":"f00b67e9b3abcbbcba68afce01050788804d463f"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/307091","number":307091,"commit":"dd2d7d33cf04fd0cb9aaac8d5915dbc4982a3c43"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/306779","number":306779,"commit":"f4146b455542142f35c50d27777d1c40ae3e4e92"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/306693","number":306693,"commit":"06daa5e6cfd246409ec19f848594d798e3826a1a"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/305973","number":305973,"commit":"43b73490a7579f4f3139b2b4aecdc3b684fab46a"}],"number":307165}
buildkite-pr-comment-->

Commit:
Fix mock typechecks

Zacqary committed:
Merge branch 'main' into dashboardApi/embeddableIntegration

nickpeihl committed:
Decouple EmbeddableContentManagementRegistry from EmbeddablePersistableStateService

Zacqary committed:
[Dashboards as Code][Embeddable API Integration] Make registry synchronous on server (#223472)

_Note: This PR does not need to be reviewed by external teams. This PR
merges into a feature branch that Kibana presentation team is working on
to support embeddable integrations with Dashboards as Code._

Convert the `EmbeddableContentManagementRegistry` back to synchronous
logic only for the server, and add an async wrapper for the public
registry.

This is achieved with some truly 12-dimensional Typescripting in
`src/platform/plugins/shared/embeddable/common/embeddable_content_management/registry.ts`

Commit:
Fix typecheck and lastSavedState example

Commit:
Revert ts-expect-errors and SavedObjectMigrationParams type, add expect-error to panel reference migration

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/308121)
* Commit: a4cef41ed6d51e75da2dd1a5d7994df968d47ecf

### Failed CI Steps
* [Check Types](https://buildkite.com/elastic/kibana-pull-request/builds/308121#01976638-0af1-4784-b818-fe985753534a)

### History
* :broken_heart: [Build #307558](https://buildkite.com/elastic/kibana-pull-request/builds/307558) failed 8adc5f437a5e89236371730128f550f5a93b1ddc
* :broken_heart: [Build #307165](https://buildkite.com/elastic/kibana-pull-request/builds/307165) failed f00b67e9b3abcbbcba68afce01050788804d463f
* :broken_heart: [Build #307091](https://buildkite.com/elastic/kibana-pull-request/builds/307091) failed dd2d7d33cf04fd0cb9aaac8d5915dbc4982a3c43
* :broken_heart: [Build #306779](https://buildkite.com/elastic/kibana-pull-request/builds/306779) failed f4146b455542142f35c50d27777d1c40ae3e4e92
* :broken_heart: [Build #306693](https://buildkite.com/elastic/kibana-pull-request/builds/306693) failed 06daa5e6cfd246409ec19f848594d798e3826a1a

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/308121","number":308121,"commit":"a4cef41ed6d51e75da2dd1a5d7994df968d47ecf"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/307558","number":307558,"commit":"8adc5f437a5e89236371730128f550f5a93b1ddc"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/307165","number":307165,"commit":"f00b67e9b3abcbbcba68afce01050788804d463f"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/307091","number":307091,"commit":"dd2d7d33cf04fd0cb9aaac8d5915dbc4982a3c43"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/306779","number":306779,"commit":"f4146b455542142f35c50d27777d1c40ae3e4e92"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/306693","number":306693,"commit":"06daa5e6cfd246409ec19f848594d798e3826a1a"}],"number":308121}
buildkite-pr-comment-->

Commit:
Revert ts-expect-error removal

Commit:
Revert "Revert ts-expect-error removal"

This reverts commit fc8f6afc00984de70e3c6f29fa0e1206f474a143.

Commit:
Revert ts-expect-error removal

Commit:
Fix typecheck

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/308485)
* Commit: 6b5a09f68bd19f5b34bc8ea2dfe45a623c28fd8f

### Failed CI Steps
* [FTR Configs #6](https://buildkite.com/elastic/kibana-pull-request/builds/308485#01976af4-114e-4443-aae5-fe7f9c0ff3c8)
* [FTR Configs #9](https://buildkite.com/elastic/kibana-pull-request/builds/308485#01976af4-1151-46fd-97a0-3ec3a35ef13b)
* [FTR Configs #10](https://buildkite.com/elastic/kibana-pull-request/builds/308485#01976af4-1151-4afe-889a-409e7d6acb39)
* [FTR Configs #11](https://buildkite.com/elastic/kibana-pull-request/builds/308485#01976af4-1152-49e9-8336-6192655fa421)
* [FTR Configs #12](https://buildkite.com/elastic/kibana-pull-request/builds/308485#01976af4-1152-4969-98d6-a13c58aa759c)
* [FTR Configs #12](https://buildkite.com/elastic/kibana-pull-request/builds/308485#01976b03-7483-4eb5-b2ba-ed0f97a682ec)
* [FTR Configs #14](https://buildkite.com/elastic/kibana-pull-request/builds/308485#01976af4-1153-4f47-840d-3d22f18fb4e4)
* [FTR Configs #37](https://buildkite.com/elastic/kibana-pull-request/builds/308485#01976af4-1167-4e18-a9c7-8782e0353bd5)
* [FTR Configs #44](https://buildkite.com/elastic/kibana-pull-request/builds/308485#01976af4-116c-4aee-80bf-6a0909bbd06c)
* [FTR Configs #63](https://buildkite.com/elastic/kibana-pull-request/builds/308485#01976af4-1177-42af-8944-aa6463ffeb65)
* [FTR Configs #87](https://buildkite.com/elastic/kibana-pull-request/builds/308485#01976af4-1184-437b-b874-e908ffa79609)
* [FTR Configs #89](https://buildkite.com/elastic/kibana-pull-request/builds/308485#01976af4-1185-430a-af58-fa02259c29d5)
* [FTR Configs #90](https://buildkite.com/elastic/kibana-pull-request/builds/308485#01976af4-1186-49ba-906a-ce83bda27b2d)
* [FTR Configs #95](https://buildkite.com/elastic/kibana-pull-request/builds/308485#01976af4-1189-4850-a440-01d0ee1c3910)
* [FTR Configs #104](https://buildkite.com/elastic/kibana-pull-request/builds/308485#01976af4-1190-45b5-ba35-49f06e661073)
* [FTR Configs #120](https://buildkite.com/elastic/kibana-pull-request/builds/308485#01976af4-1199-4a9e-b814-0c7ffd7c162d)
* [FTR Configs #122](https://buildkite.com/elastic/kibana-pull-request/builds/308485#01976af4-119a-44f5-815a-e92dc5609dbd)
* [FTR Configs #123](https://buildkite.com/elastic/kibana-pull-request/builds/308485#01976af4-119b-4573-a840-39a3e01d199f)
* [FTR Configs #124](https://buildkite.com/elastic/kibana-pull-request/builds/308485#01976af4-119c-4816-a9d6-e68ab06a5805)
* [FTR Configs #125](https://buildkite.com/elastic/kibana-pull-request/builds/308485#01976af4-119c-41a4-8cf4-6666040a67bc)
* [Synthetics @elastic/synthetics Tests](https://buildkite.com/elastic/kibana-pull-request/builds/308485#01976ada-d882-40c0-88ec-b1f3015b72e1)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/308485#01976af4-1177-42af-8944-aa6463ffeb65) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/308485/jobs/01976af4-1177-42af-8944-aa6463ffeb65/artifacts/01976b0e-ce9a-4204-a02e-f0634f5c42d4) FTR Configs #<span></span>63 / apis dashboards dashboards - create main sets panels default values
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/308485#01976af4-116f-461d-8fe4-1b3b6710a32f) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/308485/jobs/01976af4-116f-461d-8fe4-1b3b6710a32f/artifacts/01976b0f-4e90-42c5-9b98-1c761192e8f9) FTR Configs #<span></span>51 / dashboard app - group 1 edit embeddable redirects redirects via save and return button after edit
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/308485#01976af4-119c-4816-a9d6-e68ab06a5805) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/308485/jobs/01976af4-119c-4816-a9d6-e68ab06a5805/artifacts/01976b08-a88c-4799-8a3e-0daac1fb899d) FTR Configs #<span></span>124 / dashboard app - group 2 full screen mode available in view mode
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/308485#01976af4-1185-430a-af58-fa02259c29d5) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/308485/jobs/01976af4-1185-430a-af58-fa02259c29d5/artifacts/01976b0d-dadb-4663-beb2-1373c7580c90) FTR Configs #<span></span>89 / dashboard app - group 6 dashboard grid move panel Can move panel from bottom to top row
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/308485#01976af4-1189-4850-a440-01d0ee1c3910) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/308485/jobs/01976af4-1189-4850-a440-01d0ee1c3910/artifacts/01976b0d-3330-46e4-b09a-1a205d23f757) FTR Configs #<span></span>95 / Dashboard dashboard with async search not delayed should load
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/308485#01976af4-1184-437b-b874-e908ffa79609) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/308485/jobs/01976af4-1184-437b-b874-e908ffa79609/artifacts/01976b0e-9249-42a7-92d3-2f521107d26a) FTR Configs #<span></span>87 / dashboard Reporting Dashboard Reporting Screenshots Print Layout downloads a PDF file
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/308485#01976af4-1199-4a9e-b814-0c7ffd7c162d) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/308485/jobs/01976af4-1199-4a9e-b814-0c7ffd7c162d/artifacts/01976b0b-8632-4c89-b5b0-5faa381b28a4) FTR Configs #<span></span>120 / dashboard sample data dashboard should launch sample flights data set dashboard
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/308485#01976af4-1190-45b5-ba35-49f06e661073) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/308485/jobs/01976af4-1190-45b5-ba35-49f06e661073/artifacts/01976b0b-af93-441e-b7f7-2ef24340fe8d) FTR Configs #<span></span>104 / discover/embeddable discover saved search embeddable can save a search embeddable with a defined rows per page number
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/308485#01976af4-119c-41a4-8cf4-6666040a67bc) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/308485/jobs/01976af4-119c-41a4-8cf4-6666040a67bc/artifacts/01976b0a-ad50-430a-8802-78d4875b09ba) FTR Configs #<span></span>125 / Execution context Browser apps dashboard app "before all" hook in "dashboard app"
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/308485#01976af4-119b-4573-a840-39a3e01d199f) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/308485/jobs/01976af4-119b-4573-a840-39a3e01d199f/artifacts/01976b0d-e94a-43ef-be3d-684924ed5795) FTR Configs #<span></span>123 / Journey[data_stress_test_lens] Go to dashboard
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/308485#01976af4-119c-41a4-8cf4-6666040a67bc) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/308485/jobs/01976af4-119c-41a4-8cf4-6666040a67bc/artifacts/01976b0a-ad4f-4ed3-8481-707da63b02c6) FTR Configs #<span></span>125 / Journey[ecommerce_dashboard_saved_search_only] Go to Ecommerce Dashboard with Saved Search only
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/308485#01976af4-1186-49ba-906a-ce83bda27b2d) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/308485/jobs/01976af4-1186-49ba-906a-ce83bda27b2d/artifacts/01976b09-47f3-4c64-9afc-8deeac364c86) FTR Configs #<span></span>90 / lens app - group 3 lens add-to-dashboards tests should allow new lens be added by value to an existing dashboard
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/308485#01976af4-116c-4aee-80bf-6a0909bbd06c) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/308485/jobs/01976af4-116c-4aee-80bf-6a0909bbd06c/artifacts/01976b07-0488-40dd-8626-2d95be3bd5d4) FTR Configs #<span></span>44 / lens app - group 4 lens dashboard tests should hide old "explore underlying data" action
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/308485#01976af4-1167-4e18-a9c7-8782e0353bd5) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/308485/jobs/01976af4-1167-4e18-a9c7-8782e0353bd5/artifacts/01976b08-6ddb-40c8-90a4-0b33582f7f25) FTR Configs #<span></span>37 / maps app embeddable maps add-to-dashboard save flow should allow new map be added by value to a new dashboard
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/308485#01976af4-1151-46fd-97a0-3ec3a35ef13b) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/308485/jobs/01976af4-1151-46fd-97a0-3ec3a35ef13b/artifacts/01976b0d-268c-4d0e-bdb1-dae83d76a56a) FTR Configs #<span></span>9 / maps app map saved object management create should allow saving map
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/308485#01976af4-1153-4f47-840d-3d22f18fb4e4) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/308485/jobs/01976af4-1153-4f47-840d-3d22f18fb4e4/artifacts/01976b0d-0180-419c-bd22-a98558149d28) FTR Configs #<span></span>14 / Saved query management - Feature controls - v2 Security App vs Global privilege dashboard_v2 read-only privileges with savedQueryManagement.saveQuery all privilege "before all" hook for "shows read-only badge"
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/308485#01976af4-119a-44f5-815a-e92dc5609dbd) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/308485/jobs/01976af4-119a-44f5-815a-e92dc5609dbd/artifacts/01976b07-51a6-4985-88c7-2090ef36fe0e) FTR Configs #<span></span>122 / Saved query management - Feature controls Security App vs Global privilege dashboard read-only privileges with savedQueryManagement.saveQuery all privilege "before all" hook for "shows read-only badge"
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/308485#01976af4-1152-4969-98d6-a13c58aa759c) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/308485/jobs/01976af4-1152-4969-98d6-a13c58aa759c/artifacts/01976b03-5647-449f-ad5e-f5462f51b78a) FTR Configs #<span></span>12 / serverless observability UI Dataset Quality Dataset quality details overview integrations should navigate to the selected dashboard on clicking integration dashboard action 
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/308485#01976b03-7483-4eb5-b2ba-ed0f97a682ec) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/308485/jobs/01976b03-7483-4eb5-b2ba-ed0f97a682ec/artifacts/01976b0f-5539-46db-bb04-170aa6eee195) FTR Configs #<span></span>12 / serverless observability UI Dataset Quality Dataset quality details overview integrations should navigate to the selected dashboard on clicking integration dashboard action 
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/308485#01976af4-114e-4443-aae5-fe7f9c0ff3c8) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/308485/jobs/01976af4-114e-4443-aae5-fe7f9c0ff3c8/artifacts/01976b09-3720-4fb5-9baf-62ad4cddba13) FTR Configs #<span></span>6 / Visualizations - Group 2 lens app - Agg based Vis Open in Lens Pie should hide the "Convert to Lens" menu item if no split slices were defined
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/308485#01976af4-1151-4afe-889a-409e7d6acb39) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/308485/jobs/01976af4-1151-4afe-889a-409e7d6acb39/artifacts/01976b09-6786-428e-b109-d7e488002d4a) FTR Configs #<span></span>10 / Visualizations - Group 2 lens app - Agg based Vis Open in Lens Pie should hide the "Convert to Lens" menu item if no split slices were defined
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/308485#01976af4-1152-49e9-8336-6192655fa421) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/308485/jobs/01976af4-1152-49e9-8336-6192655fa421/artifacts/01976b08-2aac-40c8-ab9d-c187750b1b60) FTR Configs #<span></span>11 / Visualizations - Group 2 lens app - Agg based Vis Open in Lens Pie should hide the "Convert to Lens" menu item if no split slices were defined
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/308485#01976af4-119b-4573-a840-39a3e01d199f) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/308485/jobs/01976af4-119b-4573-a840-39a3e01d199f/artifacts/01976b0d-e949-47ee-a011-a288383e59a2) FTR Configs #<span></span>123 / Visualizations - Group 3 lens app - TSVB Open in Lens Metric should show the "Convert to Lens" menu item
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/308485#01976af4-119c-41a4-8cf4-6666040a67bc) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/308485/jobs/01976af4-119c-41a4-8cf4-6666040a67bc/artifacts/01976b0a-ad4e-4628-b398-abf6a337d8f4) FTR Configs #<span></span>125 / Visualizations - Group 3 lens app - TSVB Open in Lens Metric should show the "Convert to Lens" menu item
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/308485#01976af4-119c-4816-a9d6-e68ab06a5805) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/308485/jobs/01976af4-119c-4816-a9d6-e68ab06a5805/artifacts/01976b08-a88b-419d-87bd-acdb78356ac9) FTR Configs #<span></span>124 / Visualizations - Group 3 lens app - TSVB Open in Lens Metric should show the "Convert to Lens" menu item

### History
* :broken_heart: [Build #308450](https://buildkite.com/elastic/kibana-pull-request/builds/308450) failed 56c2a5d79477dddf49907e5b2911b18804b4edb8
* :broken_heart: [Build #308121](https://buildkite.com/elastic/kibana-pull-request/builds/308121) failed a4cef41ed6d51e75da2dd1a5d7994df968d47ecf
* :broken_heart: [Build #307558](https://buildkite.com/elastic/kibana-pull-request/builds/307558) failed 8adc5f437a5e89236371730128f550f5a93b1ddc
* :broken_heart: [Build #307165](https://buildkite.com/elastic/kibana-pull-request/builds/307165) failed f00b67e9b3abcbbcba68afce01050788804d463f
* :broken_heart: [Build #307091](https://buildkite.com/elastic/kibana-pull-request/builds/307091) failed dd2d7d33cf04fd0cb9aaac8d5915dbc4982a3c43
* :broken_heart: [Build #306779](https://buildkite.com/elastic/kibana-pull-request/builds/306779) failed f4146b455542142f35c50d27777d1c40ae3e4e92

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/308485","number":308485,"commit":"6b5a09f68bd19f5b34bc8ea2dfe45a623c28fd8f"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/308450","number":308450,"commit":"56c2a5d79477dddf49907e5b2911b18804b4edb8"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/308121","number":308121,"commit":"a4cef41ed6d51e75da2dd1a5d7994df968d47ecf"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/307558","number":307558,"commit":"8adc5f437a5e89236371730128f550f5a93b1ddc"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/307165","number":307165,"commit":"f00b67e9b3abcbbcba68afce01050788804d463f"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/307091","number":307091,"commit":"dd2d7d33cf04fd0cb9aaac8d5915dbc4982a3c43"}],"number":308485}
buildkite-pr-comment-->

Commit:
Fix binding error in registry

Commit:
Remove console.error

Zacqary commented on src/platform/plugins/shared/embeddable/common/embeddable_content_management/registry.ts:
```diff
@@ -40,16 +40,19 @@ class EmbeddableContentManagementRegistry<IsAsync extends true | false = false>
 }
 
 export class EmbeddableContentManagementRegistryPublic extends EmbeddableContentManagementRegistry<true> {
+  // @ts-expect-error Overload arrow function with member function to correctly bind `this` to superclass
```
Note that doing something like `public baseGetContentManagementDefinition` in the superclass to avoid this Typescript error will break `this` binding all over again. We really do need `getContentManagementDefinition` to be an arrow function in the superclass and then rebind it with an overload in the inheriting classes, and Typescript does NOT like this at all, but sometimes Typescript can be wrong.

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/308512)
* Commit: 0aaa869efadc616df0e828b340cb496fa56d07aa

### Failed CI Steps
* [Jest Tests #1](https://buildkite.com/elastic/kibana-pull-request/builds/308512#01976b8e-bb0e-4763-b5c8-a800f47ed6d1)
* [Jest Tests #1](https://buildkite.com/elastic/kibana-pull-request/builds/308512#01976bb4-8c35-4cdd-99c6-913397c426a5)
* [Jest Tests #9](https://buildkite.com/elastic/kibana-pull-request/builds/308512#01976b8e-bb11-426e-ba51-2a5572e5a45d)
* [FTR Configs #123](https://buildkite.com/elastic/kibana-pull-request/builds/308512#01976b8e-bb5e-4b5e-be3d-6b5ffdc1ce5d)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/308512#01976b8e-bb0e-4763-b5c8-a800f47ed6d1) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/308512/jobs/01976b8e-bb0e-4763-b5c8-a800f47ed6d1/artifacts/01976bb4-3617-4265-aca8-ab5fe946e239) Jest Tests #<span></span>1 / DashboardRenderer renders with correct options
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/308512#01976bb4-8c35-4cdd-99c6-913397c426a5) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/308512/jobs/01976bb4-8c35-4cdd-99c6-913397c426a5/artifacts/01976bba-4a2e-4a69-8eab-5db1d578efe8) Jest Tests #<span></span>1 / DashboardRenderer renders with correct options
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/308512#01976b8e-bb5e-4b5e-be3d-6b5ffdc1ce5d) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/308512/jobs/01976b8e-bb5e-4b5e-be3d-6b5ffdc1ce5d/artifacts/01976bb1-8559-44f4-abf6-f3e6ae88e1df) FTR Configs #<span></span>123 / serverless search UI Importing an existing dashboard does not show the unsaved changes badge in edit mode

### History
* :broken_heart: [Build #308485](https://buildkite.com/elastic/kibana-pull-request/builds/308485) failed 6b5a09f68bd19f5b34bc8ea2dfe45a623c28fd8f
* :broken_heart: [Build #308450](https://buildkite.com/elastic/kibana-pull-request/builds/308450) failed 56c2a5d79477dddf49907e5b2911b18804b4edb8
* :broken_heart: [Build #308121](https://buildkite.com/elastic/kibana-pull-request/builds/308121) failed a4cef41ed6d51e75da2dd1a5d7994df968d47ecf
* :broken_heart: [Build #307558](https://buildkite.com/elastic/kibana-pull-request/builds/307558) failed 8adc5f437a5e89236371730128f550f5a93b1ddc
* :broken_heart: [Build #307165](https://buildkite.com/elastic/kibana-pull-request/builds/307165) failed f00b67e9b3abcbbcba68afce01050788804d463f

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/308512","number":308512,"commit":"0aaa869efadc616df0e828b340cb496fa56d07aa"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/308485","number":308485,"commit":"6b5a09f68bd19f5b34bc8ea2dfe45a623c28fd8f"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/308450","number":308450,"commit":"56c2a5d79477dddf49907e5b2911b18804b4edb8"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/308121","number":308121,"commit":"a4cef41ed6d51e75da2dd1a5d7994df968d47ecf"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/307558","number":307558,"commit":"8adc5f437a5e89236371730128f550f5a93b1ddc"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/307165","number":307165,"commit":"f00b67e9b3abcbbcba68afce01050788804d463f"}],"number":308512}
buildkite-pr-comment-->

Commit:
Update ts-expect-error comment with github issue link

Commit:
Fix security jest test

Commit:
Merge remote-tracking branch 'upstream/main' into dashboardApi/embeddableIntegration

# Conflicts:
#	src/platform/plugins/shared/dashboard/server/content_management/v3/transform_utils.ts
#	src/platform/plugins/shared/dashboard/server/content_management/v3/transforms/out/panels_out_transforms.ts
#	src/platform/plugins/shared/embeddable/common/index.ts
#	src/platform/plugins/shared/embeddable/public/mocks.tsx
#	src/platform/plugins/shared/embeddable/public/plugin.tsx
#	src/platform/plugins/shared/embeddable/public/types.ts
#	src/platform/plugins/shared/embeddable/server/persistable_state/types.ts
#	src/platform/plugins/shared/embeddable/server/plugin.ts

Commit:
Fix bad merge

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/309072)
* Commit: 8bebf31682a7ff7f679c2b72c21e54c2a2649616

### Failed CI Steps
* [Check Types](https://buildkite.com/elastic/kibana-pull-request/builds/309072#01977a77-10d6-4627-b1a7-a0ccf709f8cb)

### History
* :broken_heart: [Build #309040](https://buildkite.com/elastic/kibana-pull-request/builds/309040) failed 9904656893a36e72df24b174b271504ee3a5d19b
* :broken_heart: [Build #308512](https://buildkite.com/elastic/kibana-pull-request/builds/308512) failed 0aaa869efadc616df0e828b340cb496fa56d07aa
* :broken_heart: [Build #308485](https://buildkite.com/elastic/kibana-pull-request/builds/308485) failed 6b5a09f68bd19f5b34bc8ea2dfe45a623c28fd8f
* :broken_heart: [Build #308450](https://buildkite.com/elastic/kibana-pull-request/builds/308450) failed 56c2a5d79477dddf49907e5b2911b18804b4edb8
* :broken_heart: [Build #308121](https://buildkite.com/elastic/kibana-pull-request/builds/308121) failed a4cef41ed6d51e75da2dd1a5d7994df968d47ecf

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/309072","number":309072,"commit":"8bebf31682a7ff7f679c2b72c21e54c2a2649616"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/309040","number":309040,"commit":"9904656893a36e72df24b174b271504ee3a5d19b"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/308512","number":308512,"commit":"0aaa869efadc616df0e828b340cb496fa56d07aa"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/308485","number":308485,"commit":"6b5a09f68bd19f5b34bc8ea2dfe45a623c28fd8f"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/308450","number":308450,"commit":"56c2a5d79477dddf49907e5b2911b18804b4edb8"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/308121","number":308121,"commit":"a4cef41ed6d51e75da2dd1a5d7994df968d47ecf"}],"number":309072}
buildkite-pr-comment-->

Commit:
Merge remote-tracking branch 'upstream/main' into dashboardApi/embeddableIntegration

# Conflicts:
#	src/platform/plugins/shared/dashboard/server/dashboard_saved_object/migrations/dashboard_saved_object_migrations.test.ts

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/309489)
* Commit: 394030a923e40e6a623881fcee775bcf8f4289f8

### Failed CI Steps
* [Check Types](https://buildkite.com/elastic/kibana-pull-request/builds/309489#01977e7f-5228-4f9a-aad1-caa709d45892)

### History
* :broken_heart: [Build #309072](https://buildkite.com/elastic/kibana-pull-request/builds/309072) failed 8bebf31682a7ff7f679c2b72c21e54c2a2649616
* :broken_heart: [Build #309040](https://buildkite.com/elastic/kibana-pull-request/builds/309040) failed 9904656893a36e72df24b174b271504ee3a5d19b
* :broken_heart: [Build #308512](https://buildkite.com/elastic/kibana-pull-request/builds/308512) failed 0aaa869efadc616df0e828b340cb496fa56d07aa
* :broken_heart: [Build #308485](https://buildkite.com/elastic/kibana-pull-request/builds/308485) failed 6b5a09f68bd19f5b34bc8ea2dfe45a623c28fd8f
* :broken_heart: [Build #308450](https://buildkite.com/elastic/kibana-pull-request/builds/308450) failed 56c2a5d79477dddf49907e5b2911b18804b4edb8

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/309489","number":309489,"commit":"394030a923e40e6a623881fcee775bcf8f4289f8"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/309072","number":309072,"commit":"8bebf31682a7ff7f679c2b72c21e54c2a2649616"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/309040","number":309040,"commit":"9904656893a36e72df24b174b271504ee3a5d19b"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/308512","number":308512,"commit":"0aaa869efadc616df0e828b340cb496fa56d07aa"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/308485","number":308485,"commit":"6b5a09f68bd19f5b34bc8ea2dfe45a623c28fd8f"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/308450","number":308450,"commit":"56c2a5d79477dddf49907e5b2911b18804b4edb8"}],"number":309489}
buildkite-pr-comment-->

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/eslint --no-cache --fix'

Commit:
Fix merge

Commit:
Merge branch 'dashboardApi/embeddableIntegration' of https://github.com/elastic/kibana into dashboardApi/embeddableIntegration

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/309543)
* Commit: c95efa7d7d91d05e990006230d9dcf84596626fe

### Failed CI Steps
* [Jest Tests #1](https://buildkite.com/elastic/kibana-pull-request/builds/309543#01977ed4-d146-4e25-915b-c604f6da29c2)
* [Jest Tests #1](https://buildkite.com/elastic/kibana-pull-request/builds/309543#01977efd-201a-4f19-8787-b170efc09260)
* [FTR Configs #120](https://buildkite.com/elastic/kibana-pull-request/builds/309543#01977ed4-d1be-4229-9c55-c9bf747edda1)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/309543#01977ed4-d1be-4229-9c55-c9bf747edda1) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/309543/jobs/01977ed4-d1be-4229-9c55-c9bf747edda1/artifacts/01977ef8-4c9c-4a9c-8923-7d95aa321d28) FTR Configs #<span></span>120 / serverless search UI Importing an existing dashboard does not show the unsaved changes badge in edit mode

### History
* :broken_heart: [Build #309489](https://buildkite.com/elastic/kibana-pull-request/builds/309489) failed 394030a923e40e6a623881fcee775bcf8f4289f8
* :broken_heart: [Build #309072](https://buildkite.com/elastic/kibana-pull-request/builds/309072) failed 8bebf31682a7ff7f679c2b72c21e54c2a2649616
* :broken_heart: [Build #309040](https://buildkite.com/elastic/kibana-pull-request/builds/309040) failed 9904656893a36e72df24b174b271504ee3a5d19b
* :broken_heart: [Build #308512](https://buildkite.com/elastic/kibana-pull-request/builds/308512) failed 0aaa869efadc616df0e828b340cb496fa56d07aa
* :broken_heart: [Build #308485](https://buildkite.com/elastic/kibana-pull-request/builds/308485) failed 6b5a09f68bd19f5b34bc8ea2dfe45a623c28fd8f

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/309543","number":309543,"commit":"c95efa7d7d91d05e990006230d9dcf84596626fe"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/309489","number":309489,"commit":"394030a923e40e6a623881fcee775bcf8f4289f8"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/309072","number":309072,"commit":"8bebf31682a7ff7f679c2b72c21e54c2a2649616"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/309040","number":309040,"commit":"9904656893a36e72df24b174b271504ee3a5d19b"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/308512","number":308512,"commit":"0aaa869efadc616df0e828b340cb496fa56d07aa"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/308485","number":308485,"commit":"6b5a09f68bd19f5b34bc8ea2dfe45a623c28fd8f"}],"number":309543}
buildkite-pr-comment-->

Commit:
Fix 8.17 panel extraction

Commit:
Fix unsaved changes flag tripping from panel reference injection

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/310172)
* Commit: f1214a6d3e6e8c71e90d1209c4407dff9037b1dd

### Failed CI Steps
* [Jest Tests #12](https://buildkite.com/elastic/kibana-pull-request/builds/310172#019784e1-2830-4d53-9de8-e6fa13d02374)
* [Jest Tests #12](https://buildkite.com/elastic/kibana-pull-request/builds/310172#01978507-1ec2-4266-962d-6b63325812fb)

### History
* :broken_heart: [Build #309543](https://buildkite.com/elastic/kibana-pull-request/builds/309543) failed c95efa7d7d91d05e990006230d9dcf84596626fe
* :broken_heart: [Build #309489](https://buildkite.com/elastic/kibana-pull-request/builds/309489) failed 394030a923e40e6a623881fcee775bcf8f4289f8
* :broken_heart: [Build #309072](https://buildkite.com/elastic/kibana-pull-request/builds/309072) failed 8bebf31682a7ff7f679c2b72c21e54c2a2649616
* :broken_heart: [Build #309040](https://buildkite.com/elastic/kibana-pull-request/builds/309040) failed 9904656893a36e72df24b174b271504ee3a5d19b
* :broken_heart: [Build #308512](https://buildkite.com/elastic/kibana-pull-request/builds/308512) failed 0aaa869efadc616df0e828b340cb496fa56d07aa

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/310172","number":310172,"commit":"f1214a6d3e6e8c71e90d1209c4407dff9037b1dd"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/309543","number":309543,"commit":"c95efa7d7d91d05e990006230d9dcf84596626fe"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/309489","number":309489,"commit":"394030a923e40e6a623881fcee775bcf8f4289f8"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/309072","number":309072,"commit":"8bebf31682a7ff7f679c2b72c21e54c2a2649616"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/309040","number":309040,"commit":"9904656893a36e72df24b174b271504ee3a5d19b"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/308512","number":308512,"commit":"0aaa869efadc616df0e828b340cb496fa56d07aa"}],"number":310172}
buildkite-pr-comment-->

Commit:
Remove embeddable.inject test from dashboard transform

Commit:
Merge remote-tracking branch 'upstream/main' into dashboardApi/embeddableIntegration

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/311067)
* Commit: e9b3286a94bb520018412c86e2eceda617c59200

### Failed CI Steps
* [Jest Tests #11](https://buildkite.com/elastic/kibana-pull-request/builds/311067#01978e9a-3abf-4555-8903-bf34dd980163)
* [Jest Tests #11](https://buildkite.com/elastic/kibana-pull-request/builds/311067#01978eba-625e-46ed-b40d-5f725b38a07f)

### History
* :broken_heart: [Build #310172](https://buildkite.com/elastic/kibana-pull-request/builds/310172) failed f1214a6d3e6e8c71e90d1209c4407dff9037b1dd
* :broken_heart: [Build #309543](https://buildkite.com/elastic/kibana-pull-request/builds/309543) failed c95efa7d7d91d05e990006230d9dcf84596626fe
* :broken_heart: [Build #309489](https://buildkite.com/elastic/kibana-pull-request/builds/309489) failed 394030a923e40e6a623881fcee775bcf8f4289f8
* :broken_heart: [Build #309072](https://buildkite.com/elastic/kibana-pull-request/builds/309072) failed 8bebf31682a7ff7f679c2b72c21e54c2a2649616
* :broken_heart: [Build #309040](https://buildkite.com/elastic/kibana-pull-request/builds/309040) failed 9904656893a36e72df24b174b271504ee3a5d19b

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/311067","number":311067,"commit":"e9b3286a94bb520018412c86e2eceda617c59200"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/310172","number":310172,"commit":"f1214a6d3e6e8c71e90d1209c4407dff9037b1dd"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/309543","number":309543,"commit":"c95efa7d7d91d05e990006230d9dcf84596626fe"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/309489","number":309489,"commit":"394030a923e40e6a623881fcee775bcf8f4289f8"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/309072","number":309072,"commit":"8bebf31682a7ff7f679c2b72c21e54c2a2649616"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/309040","number":309040,"commit":"9904656893a36e72df24b174b271504ee3a5d19b"}],"number":311067}
buildkite-pr-comment-->

Commit:
Remove inject tests and fix async error test in panelOutTransforms

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/311107)
* Commit: 799a7f6dfb03f55ebf2e2775aecfff9dc171ab17

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/33a6307b116280647b84f338734f8839410ec2bc) | [after](https://github.com/elastic/kibana/commit/799a7f6dfb03f55ebf2e2775aecfff9dc171ab17) | diff |
| --- | --- | --- | --- |
| `embeddable` | 107 | 108 | +1 |

#### Public APIs missing comments
> Total count of every public API that lacks a comment. Target amount is 0. Run `node scripts/build_api_docs --plugin [yourplugin] --stats comments` for more detailed information.


| id | [before](https://github.com/elastic/kibana/commit/33a6307b116280647b84f338734f8839410ec2bc) | [after](https://github.com/elastic/kibana/commit/799a7f6dfb03f55ebf2e2775aecfff9dc171ab17) | diff |
| --- | --- | --- | --- |
| `dashboard` | 95 | 103 | +8 |
| `embeddable` | 124 | 134 | +10 |
| total |  |  | +18 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/33a6307b116280647b84f338734f8839410ec2bc) | [after](https://github.com/elastic/kibana/commit/799a7f6dfb03f55ebf2e2775aecfff9dc171ab17) | diff |
| --- | --- | --- | --- |
| `apm` | 2.6MB | 2.6MB | +10.0B |
| `dashboard` | 618.5KB | 618.9KB | +395.0B |
| `securitySolution` | 9.4MB | 9.4MB | +5.0B |
| total |  |  | +410.0B |

#### Public APIs missing exports
> Total count of every type that is part of your API that should be exported but is not. This will cause broken links in the API documentation system. Target amount is 0. `Run node scripts/build_api_docs --plugin [yourplugin] --stats exports` for more detailed information.


| id | [before](https://github.com/elastic/kibana/commit/33a6307b116280647b84f338734f8839410ec2bc) | [after](https://github.com/elastic/kibana/commit/799a7f6dfb03f55ebf2e2775aecfff9dc171ab17) | diff |
| --- | --- | --- | --- |
| `dashboard` | 12 | 13 | +1 |
| `embeddable` | 1 | 2 | +1 |
| total |  |  | +2 |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/33a6307b116280647b84f338734f8839410ec2bc) | [after](https://github.com/elastic/kibana/commit/799a7f6dfb03f55ebf2e2775aecfff9dc171ab17) | diff |
| --- | --- | --- | --- |
| `embeddable` | 16.7KB | 17.6KB | +927.0B |

<details><summary>Unknown metric groups</summary>

#### API count
| id | [before](https://github.com/elastic/kibana/commit/33a6307b116280647b84f338734f8839410ec2bc) | [after](https://github.com/elastic/kibana/commit/799a7f6dfb03f55ebf2e2775aecfff9dc171ab17) | diff |
| --- | --- | --- | --- |
| `dashboard` | 100 | 108 | +8 |
| `embeddable` | 148 | 158 | +10 |
| total |  |  | +18 |

#### ESLint disabled line counts
| id | [before](https://github.com/elastic/kibana/commit/33a6307b116280647b84f338734f8839410ec2bc) | [after](https://github.com/elastic/kibana/commit/799a7f6dfb03f55ebf2e2775aecfff9dc171ab17) | diff |
| --- | --- | --- | --- |
| `embeddable` | 1 | 2 | +1 |

#### References to deprecated APIs
| id | [before](https://github.com/elastic/kibana/commit/33a6307b116280647b84f338734f8839410ec2bc) | [after](https://github.com/elastic/kibana/commit/799a7f6dfb03f55ebf2e2775aecfff9dc171ab17) | diff |
| --- | --- | --- | --- |
| `@kbn/saved-search-component` | 0 | 1 | +1 |
| `aiops` | 8 | 11 | +3 |
| `controls` | 18 | 20 | +2 |
| `dashboard` | 5 | 14 | +9 |
| `dataVisualizer` | 11 | 12 | +1 |
| `discover` | 9 | 16 | +7 |
| `embeddableEnhanced` | 0 | 3 | +3 |
| `imageEmbeddable` | 2 | 3 | +1 |
| `lens` | 76 | 86 | +10 |
| `links` | 1 | 3 | +2 |
| `maps` | 26 | 37 | +11 |
| `slo` | 3 | 4 | +1 |
| `synthetics` | 26 | 27 | +1 |
| `visualizations` | 53 | 62 | +9 |
| total |  |  | +61 |

#### Total ESLint disabled count
| id | [before](https://github.com/elastic/kibana/commit/33a6307b116280647b84f338734f8839410ec2bc) | [after](https://github.com/elastic/kibana/commit/799a7f6dfb03f55ebf2e2775aecfff9dc171ab17) | diff |
| --- | --- | --- | --- |
| `embeddable` | 1 | 2 | +1 |


</details>


### History
* :broken_heart: [Build #311067](https://buildkite.com/elastic/kibana-pull-request/builds/311067) failed e9b3286a94bb520018412c86e2eceda617c59200
* :broken_heart: [Build #310172](https://buildkite.com/elastic/kibana-pull-request/builds/310172) failed f1214a6d3e6e8c71e90d1209c4407dff9037b1dd
* :broken_heart: [Build #309543](https://buildkite.com/elastic/kibana-pull-request/builds/309543) failed c95efa7d7d91d05e990006230d9dcf84596626fe
* :broken_heart: [Build #309489](https://buildkite.com/elastic/kibana-pull-request/builds/309489) failed 394030a923e40e6a623881fcee775bcf8f4289f8
* :broken_heart: [Build #309072](https://buildkite.com/elastic/kibana-pull-request/builds/309072) failed 8bebf31682a7ff7f679c2b72c21e54c2a2649616

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/311107","number":311107,"commit":"799a7f6dfb03f55ebf2e2775aecfff9dc171ab17"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/311067","number":311067,"commit":"e9b3286a94bb520018412c86e2eceda617c59200"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/310172","number":310172,"commit":"f1214a6d3e6e8c71e90d1209c4407dff9037b1dd"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/309543","number":309543,"commit":"c95efa7d7d91d05e990006230d9dcf84596626fe"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/309489","number":309489,"commit":"394030a923e40e6a623881fcee775bcf8f4289f8"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/309072","number":309072,"commit":"8bebf31682a7ff7f679c2b72c21e54c2a2649616"}],"number":311107}
buildkite-pr-comment-->

nreese commented:
@elasticmachine merge upstream

elasticmachine committed:
Merge branch 'main' into dashboardApi/embeddableIntegration

elasticmachine commented:
<!-- buildkite-pr-bot-trigger-comment -->:robot: Jobs for this PR can be triggered through checkboxes. :construction: 

:information_source: To trigger the CI, please tick the checkbox below :point_down: 

- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-pull-request --> Click to trigger **kibana-pull-request** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-project-from-pr --> Click to trigger **kibana-deploy-project-from-pr** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-cloud-from-pr --> Click to trigger **kibana-deploy-cloud-from-pr** for this PR!

nreese commented:
Sorry for the pings, accidentally clicked "Ready for review". PR is not ready for review and is still in draft. Please "unsubscribe" to avoid getting updates for this PR.

nreese committed:
Merge branch 'main' into dashboardApi/embeddableIntegration

nreese commented:
@elasticmachine merge upstream

elasticmachine committed:
Merge branch 'main' into dashboardApi/embeddableIntegration
